### PR TITLE
aarch64_cpu_sve: fix issues

### PR DESCRIPTION
**Pull request by dzhengfy**
Some arm hardwares do not support sve512 and sve256. So we should handle this special condition.

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
Before the fix:
```
[rhel.aarch64_cpu_sve.positive_test.boot_test.enable_sve]
TestFail: Expect guest support sve&#10;

```
After the fix:
` (1/9) type_specific.io-github-autotest-libvirt.aarch64_cpu_sve.positive_test.boot_test.enable_sve: PASS (51.02 s)`

**Comment by dzhengfy** [comment]
@hs0210 Could you help review?

File: libvirt/tests/src/cpu/aarch64_cpu_sve.py [hunk 1]
```diff
@@ -97,9 +97,9 @@ def _get_maxium_sve_length(vm):
             sve_length_byte = re.search(r"length (\d+) bytes", ret).groups()[0]
             # Change max_length into sve + length(bit) E.g. sve512
             sve_length_bit = "sve" + str(int(sve_length_byte) * 8)
-            logging.debug("guest sve_length_bit is %s" % sve_length_bit)
+            logging.debug("The guest maximum SVE vector length is %s" % sve_length_bit)
```
Comment (Yingshun) [inline libvirt/tests/src/cpu/aarch64_cpu_sve.py:100]:
```suggestion
            logging.debug("The guest maximum SVE vector length is %s", sve_length_bit)
```

**Comment by hs0210** [comment]
LGTM
```
# /usr/local/bin/avocado run --vt-type libvirt --test-runner=runner --vt-machine-type arm64-mmio aarch64_cpu_sve --vt-connect-uri qemu:///system
JOB ID     : a67c914c374395967e9e48750e90ef8d37c5e70a
JOB LOG    : /root/avocado/job-results/job-2024-02-28T02.17-a67c914/job.log
(1/9) type_specific.io-github-autotest-libvirt.aarch64_cpu_sve.positive_test.boot_test.enable_sve: PASS (82.08 s)
(2/9) type_specific.io-github-autotest-libvirt.aarch64_cpu_sve.positive_test.boot_test.disable_sve: PASS (78.36 s)
(3/9) type_specific.io-github-autotest-libvirt.aarch64_cpu_sve.positive_test.vector_length_test.valid_length.single_vector.sve128: PASS (81.83 s)
(4/9) type_specific.io-github-autotest-libvirt.aarch64_cpu_sve.positive_test.vector_length_test.valid_length.single_vector.sve256: PASS (81.76 s)
(5/9) type_specific.io-github-autotest-libvirt.aarch64_cpu_sve.positive_test.vector_length_test.valid_length.single_vector.sve512: PASS (82.65 s)
(6/9) type_specific.io-github-autotest-libvirt.aarch64_cpu_sve.positive_test.vector_length_test.valid_length.mutiple_vector:PASS (81.76 s)
(7/9) type_specific.io-github-autotest-libvirt.aarch64_cpu_sve.negative_test.vector_length_test.invalid_length: PASS (46.76 s)
(8/9) type_specific.io-github-autotest-libvirt.aarch64_cpu_sve.negative_test.vector_length_test.conflict_length: PASS (46.86 s)
(9/9) type_specific.io-github-autotest-libvirt.aarch64_cpu_sve.negative_test.no_host_sve_support: CANCEL: Host supports SVE (45.30 s)
RESULTS    : PASS 8 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /root/avocado/job-results/job-2024-02-28T02.17-a67c914/results.html
JOB TIME   : 628.72 s
```

File: libvirt/tests/src/cpu/aarch64_cpu_sve.py [hunk 2]
```diff
@@ -170,6 +170,9 @@ def _guest_has_sve(vm):
                 test.error("Failed to define guest: %s" % str(e))
 
         result = virsh.start(vm_name)
+        special_error_msg = "CPU does not support the vector length"
+        if re.search(special_error_msg, result.stderr_text):
```
Comment (chunfuwen) [inline libvirt/tests/src/cpu/aarch64_cpu_sve.py:174]: Do we have any virsh command output or native linux command to check vector feature before call vm.start()?

**Comment by dzhengfy** [comment]
@hs0210 Could you help review?

**Review by hs0210 — APPROVED** [review]

File: libvirt/tests/src/cpu/aarch64_cpu_sve.py (hunk 2 above)
Comment (dzhengfy) [reply libvirt/tests/src/cpu/aarch64_cpu_sve.py:174]: @chunfuwen seems not yet. We have to start vm and check the file within the vm. But if I know a better new way, I would like to update it

File: libvirt/tests/src/cpu/aarch64_cpu_sve.py (hunk 1 above)
Comment (dzhengfy) [reply libvirt/tests/src/cpu/aarch64_cpu_sve.py]: updated, thanks

**Review by chunfuwen — APPROVED** [review]
lgtm

**Pull request merged**